li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/consumer/LiKafkaFederatedConsumerImpl.java

```diff
@@ -771,4 +818,37 @@ private void ensureOpen() {
       throw new IllegalStateException("this federated consumer has already been closed");
     }
   }
+
+  private void maybeResubscribe() {
+    if (_closed || _currentSubscription.getSubscriptionType() == FederatedSubscriptionState.SubscriptionType.NONE ||
+        _currentSubscription.getTopicsWaitingToBeCreated().isEmpty()) {
+      return;
+    }
+
+    boolean topicCreated = false;
+    for (String topic : _currentSubscription.getTopicsWaitingToBeCreated()) {
+      try {
+        if (_mdsClient.getClusterForTopic(topic, _clusterGroup, _mdsRequestTimeoutMs) != null) {
+          topicCreated = true;
+          break;
+        }
+      } catch (MetadataServiceClientException e) {
+        LOG.warn("cannot get cluster info for topic {}. ignoring...", topic);
+      }
+    }
+
+    if (!topicCreated) {
+      return;
+    }
+
+    // Re-assign/subscribe
+    switch (_currentSubscription.getSubscriptionType()) {
+      case USER_ASSIGNED:
+        assign(_currentSubscription.getAssignment());
+        break;
+
+      default:
+        throw new IllegalStateException("Unsupportd subscription type");
```

Comment:
you want to log here as this will throw on a thread out of the pool, which we may not see

Reply:
done